soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -251,9 +237,24 @@ void z_mp_entry(void)
 	/* Interrupt must be enabled while running on current core */
 	irq_enable(DT_IRQN(DT_INST(0, intel_cavs_idc)));
 
-#ifdef CONFIG_SMP_BOOT_DELAY
-	cavs_idc_smp_init(NULL);
-#endif
+	/* Unfortunately the interrupt controller doesn't understand
+	 * that each CPU has its own mask register (the timer has a
+	 * similar hook).  Needed only on hardware with ROMs that
+	 * disable this; cAVS 2.5 starts with an unmasked hardware
+	 * default.
+	 */
+	if (!IS_ENABLED(CONFIG_SOC_SERIES_INTEL_CAVS_V25)) {
+		CAVS_INTCTRL[start_rec.cpu].l2.clear = CAVS_L2_IDC;
+	}
+
+	/* Unmask IDC interrupts from this core to all others.  A
+	 * delay is needed following the write on older hardware, or
+	 * else the modification gets lots.  Voodoo.
```

Comment:
Sorry, "gets lost".  Will fix the typo :)

And I don't have an explanation here.  I stumbled on this spot while porting, but the delay really is required.

Reply:
Oh, and it's not as simple as just a short delay here.  The core does indeed come up just fine; e.g. in the tests/kernel/smp_boot_delay test I added, which duplicates as much as possible (with the exception that it's triggered off of timer code on core 0, since we have nothing to send an IPC) the SOF core bringup flow.